src/views/Appointments/Booking.vue

```diff
@@ -188,12 +188,14 @@ export default {
 			this.slots = []
 			this.loadingSlots = true
 
-			const startOfDay = new Date(this.selectedDate.getTime())
+			const selectedDay = this.selectedDate.getFullYear().toString() + '-' + 
+								(this.selectedDate.getMonth() + 1).toString() + '-' + 
+							    this.selectedDate.getDate().toString()
```

Comment:
We need proper formatting here. Month and date might only have a single digit.

Please use the `getYYYYMMDDFromDate(date)` from `src/utils/date.js` which already implements formatting as `YYYY-MM-DD`.

Reply:
```js
import { getYYYYMMDDFromDate } from '../../utils/date.js'

// [...] further down

const selectedDay = getYYYYMMDDFromDate(this.selectedDate)
```

Reply:
Yeah, that is why Georg reverts the shift. See `date.getTime() - (date.getTimezoneOffset() * 60000))`

I tested it and it should work correctly.

Reply:
Yeah, I don't trust the calculation in getYYYYMMDDFromDate

Tested with:

- Value In
Wed Nov 27 2024 14:44:54 GMT-0500 (Eastern Standard Time)

- Value produced by "new Date(date.getTime() - date.getTimezoneOffset() * 60000)"
Wed Nov 27 2024 09:44:54 GMT-0500 (Eastern Standard Time)

As you can see it shifted the time by 5 hours. This would have the effect of shifting the date, if the input value time was less then the offset.

E.g.
Input 
Wed Nov 27 2024 2:00:00 GMT-0500 (Eastern Standard Time)

Result
Wed Nov 26 2024 21:00:00 GMT-0500 (Eastern Standard Time)

Then toISOString produces this 
2024-11-27T14:44:54.790Z

The end result seems correct in my browser and time zone but will this be correct for every time zone and browser. IDK

Reply:
I get that it is confusing. The thing is, that JavaScript's Date is not time zone aware. It is **always** in the local time zone, regardless which time zone you pass in the constructor.

We have unit tests in place to ensure that this implementation works and it is used throughout Calendar.

If it breaks we can fix it. The important thing is that we have one central implementation and not many copied across the code base because this is way harder to maintain.

Reply:
See also https://stackoverflow.com/a/29774197/7840137

Reply:
But yeah, let's not get pedantic over 3 lines of code.